syntax = "proto3";

package vectordb;

service VectorDBService {
    rpc Init (InitRequest) returns (InitResponse);
    rpc UpsertVector (UpsertRequest) returns (UpsertResponse);
    rpc QueryVector (QueryRequest) returns (QueryResponse);
}


message InitRequest {
    optional string db_path = 1;
    int32 vector_size = 2;
    string metric = 3;
    string dest_path = 4;
}

message InitResponse {
    int32 code = 1;
    string message = 2;
}

message UpsertResponse {
  int32 code = 1;
}

message QueryResponse {
    int32 code = 1;
  repeated int32 matched_vector_ids = 2;
}

message UpsertRequest {
    int32 vector_id = 1;
    repeated float vector = 2;
}

message QueryRequest {
    repeated float vector = 1;
    int32 n = 2;
}

